parquet-column/src/main/java/org/apache/parquet/column/values/bloomfilter/BloomFilter.java

```diff
@@ -0,0 +1,373 @@
+ * each tiny Bloom filter. Each tiny Bloom filter is 32 bytes to take advantage of 32-byte SIMD
+ * instruction.
+ */
+
+public class BloomFilter {
+  // Bloom filter Hash strategy .
+  public enum HashStrategy {
+    MURMUR3_X64_128,
+  }
+
+  // Bloom filter algorithm.
+  public enum Algorithm {
+    BLOCK,
+  }
+
+  // Bytes in a tiny Bloom filter block.
+  private static final int BYTES_PER_FILTER_BLOCK = 32;
+
+  // Default seed for hash function, it comes from System.nanoTime().
+  private static final int DEFAULT_SEED = 1361930890;
+
+  // Minimum Bloom filter size, set to size of a tiny Bloom filter block
+  public static final int MINIMUM_BLOOM_FILTER_BYTES = 32;
+
+  // Maximum Bloom filter size, it sets to default HDFS block size for upper boundary check
+  // This should be re-consider when implementing write side logic.
+  public static final int MAXIMUM_BLOOM_FILTER_BYTES = 128 * 1024 * 1024;
+
+  // The number of bits to set in a tiny Bloom filter
+  private static final int BITS_SET_PER_BLOCK = 8;
+
+  // The header of Bloom filter, it includes number of bytes, algorithm and hash enumeration.
+  public static final int HEADER_SIZE = 12;
+
+  // Hash strategy used in this Bloom filter.
+  public final HashStrategy hashStrategy;
+
+  // Algorithm used in this Bloom filter.
+  public final Algorithm algorithm;
+
+  // The underlying byte array for Bloom filter bitset.
+  private byte[] bitset;
+
+  // A integer array buffer of underlying bitset to help setting bits.
+  private IntBuffer intBuffer;
+
+  // Hash function use to compute hash for column value.
+  private HashFunction hashFunction;
+
+  // The block-based algorithm needs 8 odd SALT values to calculate eight index
+  // of bit to set, one bit in 32-bit word.
+  private static final int SALT[] = {0x47b6137b, 0x44974d91, 0x8824ad5b, 0xa2b7289d,
+    0x705495c7, 0x2df1424b, 0x9efc4947, 0x5c6bfb31};
+
+  /**
+   * Constructor of Bloom filter.
+   *
+   * @param numBytes The number of bytes for Bloom filter bitset. The range of num_bytes should be within
+   *                 [MINIMUM_BLOOM_FILTER_BYTES, MAXIMUM_BLOOM_FILTER_BYTES], it will be rounded up/down
+   *                 to lower/upper bound if num_bytes is out of range and also will rounded up to a power
+   *                 of 2. It uses murmur3_x64_128 as its default hash function and block-based algorithm
+   *                 as default algorithm.
+   */
+  public BloomFilter(int numBytes) {
+    this(numBytes, HashStrategy.MURMUR3_X64_128, Algorithm.BLOCK);
+  }
+
+  /**
+   * Constructor of Bloom filter. It uses murmur3_x64_128 as its default hash
+   * function and block-based algorithm as its default algorithm.
+   *
+   * @param numBytes The number of bytes for Bloom filter bitset
+   * @param hashStrategy The hash strategy of Bloom filter.
+   * @param algorithm The algorithm of Bloom filter.
+   */
+  private BloomFilter(int numBytes, HashStrategy hashStrategy, Algorithm algorithm) {
```

Comment:
Thanks @jbapple-cloudera for reviewing.

This patch makes some confusion here. It is just a early patch to show how a Bloom filter is integrated with ParquetFileReader and ParuqetFileWriter as you mentioned in vote email. With this we can commit parquet-format patch firstly.  Since without  parquet-format patch, the reader and writer implementation cannot pass the build.

Reply:
That works for me.